packages/tangle/images/TestBookerMarkerMappings.md

```diff
@@ -0,0 +1,236 @@
+# Test of Booker and Marker Mapping
+
+
+The Marker tool is tested for the booker with the following scenario:
+
+![](https://imgur.com/sCkXXrG.png)
+
+
+## General rules
+
+We apply the following defintions:
+- a colored field indicates that a value is set
+- a field without color indicates the values are inherited
+- [ . ] for a BranchID indicates inheritance. For a data message or transaction that inherits a branchID there is no action/update of mapping  necessary.
+- Data messages always have Payload.BranchID = Main
+
+The transaction-by-transaction development of the data associated with the marker DAG is as follows:
+
+# 
+## Step-by-step analysis
+### Transaction 1 (create Marker): 
+
+- Transaction 1: 
+    - create sequence ID 1, for now sequence ID of Main = 1
+    - PastMarker.BranchID = MainBranchID
+    - Payload.BranchID = MainBranchID
+
+Note that MainBranchID does not own sequence ID 1, although the PastMarker has sequence ID 1.
+
+![](https://imgur.com/Wsobgv0.png)
+
+### Transaction 2 (create Marker, double spend): 
+
+- Transacion 2 double spends transaction 1
+- Transaction 1:
+    - create Branch A
+    - Branch A gets sequence ID = 1
+    - PastMarker.BranchID = A
+    - Payload.BranchID = A
+- Transaction 2: 
+    - create Branch B
+    - create sequence ID 2
+    - PastMarker.BranchID = B
+    - Payload.BranchID = B
+
+![](https://imgur.com/9SqyjlY.png)
+
+### Transaction 3 (create Marker, double spend): 
+
+- Transaction 3 double spends transaction 1
+- Transaction 3 
+    - create Branch C
+    - create sequence ID 3
+    - PastMarker.BranchID = C
+    - Payload.BranchID = C
+
+![](https://imgur.com/N8xffwD.png)
+
+### Data message 4 (create Marker): 
+
+- Data message 4
+    - create sequence ID 4, for now sequence ID of Main = 4
+    - PastMarker.BranchID = Main
+    - Payload.BranchID = Main (since data)
+
+![](https://imgur.com/GIJjo0b.png)
+
+### Transaction 5 (create Marker): 
+
+- Transaction 5
+    - PastMarker.BranchID = Main
+    - Payload.BranchID = Main
+
+![](https://imgur.com/YMGAbiF.png)
+
+### Transaction 6 (create Marker): 
+
+- Transaction 6
+    - PastMarker.BranchID = [A]
+    - Payload.BranchID = Main, since no double spend
```

Comment:
I might be wrong but according to the UTXO dag tx6 spends `Output G` which is in `Branch A`. Hence, the payload should also be in that branch. But it doesn't really change anything and updating all the figures might be too much of a hassle.

Reply:
ups. ok. i made a note in the text though.